src/js/d3ViewComponent.js

```diff
@@ -0,0 +1,94 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    // The D3 view component that is able to:
+    // 1. Convert jQuery DOM elements to D3 elements;
+    // 2. Attach D3 DOM event listeners;
+    // 3. Synthesize that.options.styles and that.options.selectors to combine elements with the same key into that.classes
+
+    fluid.defaults("gpii.d3ViewComponent", {
+        gradeNames: ["fluid.viewComponent", "autoInit"],
+        members: {
+            classes: {
+                expander: {
+                    funcName: "gpii.d3ViewComponent.synthesizeClasses",
+                    args: ["{that}.options.styles", "{that}.options.selectors"]
+                }
+            }
+        },
+        invokers: {
+            jQueryToD3: {
```

Comment:
What if instead we made it easier to get a d3 object. For example add a feature so that `{that}.d3.selectName` was possible, and maybe provide `that.d3Container` and `that.locate.d3("selectorName")`. We might have talked about this before, but I'm thinking that it would be faster than having to call jQueryToD3 every time, and function could expect to get d3 elements passed in instead of having to convert them all the time. 

And just to be clear, all the typical methods of a viewComponent would still persist. So accessing the jQuery equivalents would still be possible in the typical way.